esxi/tests/test_unit.py

```diff
@@ -660,3 +660,326 @@ def test_runtime_host_not_found(vcsim_instance, dd_run_check, caplog, service_in
 
     dd_run_check(check)
     assert "Missing runtime.host details for VM vm1" in caplog.text
+
+
+@pytest.mark.parametrize(
+    "resource_filters, expected_vms",
+    [
+        pytest.param(
+            [
+                {
+                    'type': 'include',
+                    'resource': 'vm',
+                    'property': 'name',
+                    'patterns': [
+                        'vm1.*',
+                    ],
+                }
+            ],
+            ["vm1"],
+            id="include list- name",
+        ),
+        pytest.param(
+            [
+                {
+                    'type': 'exclude',
+                    'resource': 'vm',
+                    'patterns': [
+                        'vm1.*',
+                    ],
+                }
+            ],
+            ["vm2"],
+            id="exclude list- name",
+        ),
+        pytest.param(
+            [
+                {
+                    'type': 'exclude',
+                    'resource': 'vm',
+                    'property': 'name',
+                    'patterns': [
+                        'vm.*',
+                    ],
+                }
+            ],
+            [],
+            id="all excluded- name",
+        ),
+        pytest.param(
+            [
+                {
+                    'type': 'exclude',
+                    'resource': 'vm',
+                    'property': 'name',
+                    'patterns': [
+                        'test.*',
+                    ],
+                }
+            ],
+            ["vm1", "vm2"],
+            id="none excluded- name",
+        ),
+        pytest.param(
+            [
+                {
+                    'type': 'include',
+                    'resource': 'vm',
+                    'property': 'name',
+                    'patterns': [
+                        'vm1.*',
+                    ],
+                },
+                {
+                    'type': 'include',
+                    'resource': 'vm',
+                    'property': 'name',
+                    'patterns': [
+                        'test.*',
+                    ],
+                },
+            ],
+            ["vm1"],
+            id="unrecognized pattern- name",
```

Comment:
What is the goal of this test

Reply:
To test the case where a filter does not match any of the resources

Reply:
sure, I can do that to be more clear!